apps/microtvm/zephyr_cmsisnn/src/main.c

```diff
@@ -34,7 +34,7 @@ extern float output_storage[12];
 
 extern const size_t output_len;
 
-static uint8_t g_crt_workspace[TVMGEN_DEFAULT_WORKSPACE_SIZE + 256];
+static uint8_t g_crt_workspace[TVMGEN_DEFAULT_WORKSPACE_SIZE + 512];
```

Comment:
Did I add this? This looks like a hack to increase the workspace size for over-allocation, if we still need this is something broken in USMP?

Reply:
Yea -- I was puzzled by this too. 

In fact, USMP is not enabled for cmsis-nn (yet -- its coming in the next PR) and this is only needed for Zephyr -- other cmsis-nn tests are fine. I am not familiar with Zephyr related impacts here.

Reply:
i.e. when USMP is enabled we dont need this workspace altogether.

Reply:
This was surfaced when I removed this : https://github.com/apache/tvm/blob/c54a3dd6999ead14fd118e6d84f2b64a1ecf3b9d/python/tvm/micro/model_library_format.py#L217-L219.

Which seems to be adding the workspace again for external_codegens.

I think we need investigate this further

Reply:
Zephyr should have no real impact on the memory pre-allocated here as it's just a block in flash, this is deeply worrying as the allocator is configured here:

https://github.com/apache/tvm/blob/7d831e1dd34d441e9217096d0baf78e6f97c77d9/apps/microtvm/zephyr_cmsisnn/src/main.c#L62

Thus the allocator itself should never go over if it's performing properly, something is very weird here but I agree we should investigate further when we've integrated USMP fully.